components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -262,4 +267,91 @@ private String logMessageDetails(MessageContext messageContext, long elapsedTime
+            consumerKey = authContext.getConsumerKey();
+            username = authContext.getUsername();
+            applicationName = authContext.getApplicationName();
+            applicationId = authContext.getApplicationId();
+        }
+
+        String context = (String) messageContext.getProperty(RESTConstants.REST_API_CONTEXT);
+        String api_version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API);
+
+        int index = api_version.indexOf("--");
+
+        if (index != -1) {
+            api_version = api_version.substring(index + 2);
+        }
+
+        String api = api_version.split(":")[0];
+        String version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API_VERSION);
+
+        String fullRequestPath = (String) messageContext.getProperty(RESTConstants.REST_FULL_REQUEST_PATH);
+        int tenantDomainIndex = fullRequestPath.indexOf("/t/");
+        String apiPublisher = (String) messageContext.getProperty(APIMgtGatewayConstants.API_PUBLISHER);
+        String tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        if (tenantDomainIndex != -1) {
+            String temp = fullRequestPath.substring(tenantDomainIndex + 3, fullRequestPath.length());
+            tenantDomain = temp.substring(0, temp.indexOf("/"));
+        }
+
+        if (apiPublisher == null) {
+            apiPublisher = getAPIProviderFromRESTAPI(api_version);
+        }
+
+        if (apiPublisher != null && !apiPublisher.endsWith(tenantDomain)) {
+            apiPublisher = apiPublisher + "@" + tenantDomain;
+        }
+
+        String resource = extractResource(messageContext);
+        String method = (String) (axis2MsgContext.getProperty(
+                Constants.Configuration.HTTP_METHOD));
+        String hostName = APIUtil.getHostAddress();
+
+        messageContext.setProperty(APIMgtGatewayConstants.CONSUMER_KEY, consumerKey);
+        messageContext.setProperty(APIMgtGatewayConstants.USER_ID, username);
+        messageContext.setProperty(APIMgtGatewayConstants.CONTEXT, context);
+        messageContext.setProperty(APIMgtGatewayConstants.API_VERSION, api_version);
+        messageContext.setProperty(APIMgtGatewayConstants.API, api);
+        messageContext.setProperty(APIMgtGatewayConstants.VERSION, version);
+        messageContext.setProperty(APIMgtGatewayConstants.RESOURCE, resource);
+        messageContext.setProperty(APIMgtGatewayConstants.HTTP_METHOD, method);
+        messageContext.setProperty(APIMgtGatewayConstants.HOST_NAME, hostName);
+        messageContext.setProperty(APIMgtGatewayConstants.API_PUBLISHER, apiPublisher);
+        messageContext.setProperty(APIMgtGatewayConstants.APPLICATION_NAME, applicationName);
+        messageContext.setProperty(APIMgtGatewayConstants.APPLICATION_ID, applicationId);
+    }
+
+    private String extractResource(MessageContext mc) {
+        String resource = "/";
+        Pattern pattern = Pattern.compile("^/.+?/.+?([/?].+)$");
```

Comment:
Fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/68b378607966da79d994692e05c46be213f2ecc0)